### DISCUSSION TITLE
Custom progressbar column

### DISCUSSION THREAD

**Discussion by Liteolika** [spectreconsole/spectre.console#522] [Q&A] [answered]
Is there any way to extend/create a custom progressbar column?

**Comment by patriksvensson** [comment, marked as answer]
Yes. I'm not currently at a computer, but you should be able to by subclassing the `ProgressColumn` class.

**Reply by Liteolika** [reply]
Ill try again. I got some inheritance issues when i tried last but i didnt dig deeper :) Thanks Patrik.

**Reply by patriksvensson** [reply]
I think it would be the easiest to look at one of the existing columns, like this one: https://github.com/spectreconsole/spectre.console/blob/main/src/Spectre.Console/Live/Progress/Columns/PercentageColumn.cs

**Reply by phil-scott-78** [reply]
Additionally to what Patrik pointed to, here's a custom column I've created outside of the Spectre package 
 https://github.com/phil-scott-78/dotnet-soddi/blob/master/src/Soddi/ProgressBar/TorrentProgressBarColumn.cs

**Reply by Liteolika** [reply]
Thanks! I will give it a try.
